File: intellij-community/platform/lang-impl/src/com/intellij/largeFilesEditor/search/Position.java
// Copyright 2000-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
package com.intellij.largeFilesEditor.search;

public class Position {
  public long pageNumber;
  public int symbolOffsetInPage;

  public Position() {
    this.pageNumber = 0;
    this.symbolOffsetInPage = 0;
  }

  public Position(long pageNumber, int symbolOffsetInPage) {
    this.pageNumber = pageNumber;
    this.symbolOffsetInPage = symbolOffsetInPage;
  }

  public void reset(long pageNumber, int symbolNumber) {
    this.pageNumber = pageNumber;
    this.symbolOffsetInPage = symbolNumber;
  }

  @Override
  public boolean equals(Object target) {
    if (this == target) {
      return true;
    }

    if (target instanceof Position) {
      if (pageNumber == ((Position)target).pageNumber
          && symbolOffsetInPage == ((Position)target).symbolOffsetInPage) {
        return true;
      }
    }
    return false;
  }
}

File: plugins/formal-verification/src/org/jetbrains/kotlin/formver/scala/silicon/ast/Position.kt
/*
 * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
 * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
 */

package org.jetbrains.kotlin.formver.scala.silicon.ast

import org.jetbrains.kotlin.formver.scala.IntoViper
import viper.silver.ast.`NoPosition$`

sealed class Position : IntoViper<viper.silver.ast.Position> {
    data object NoPosition : Position() {
        override fun toViper(): viper.silver.ast.Position = `NoPosition$`.`MODULE$`
    }

    class LineColumnPosition(private val line: Int, private val column: Int) : Position() {
        override fun toViper(): viper.silver.ast.Position = viper.silver.ast.LineColumnPosition(line, column)
    }
}
